pkg/api/api.go

```diff
@@ -2313,6 +2313,35 @@ func (s *ScalewayAPI) AttachIP(ipID, serverID string) error {
 	return err
 }
 
+// DetachIP detaches an IP from a server
+func (s *ScalewayAPI) DetachIP(ipID string) error {
+	var update struct {
+		Address      string `json:"address"`
+		ID           string `json:"id"`
+		Organization string `json:"organization"`
+	}
+
+	ip, err := s.GetIP(ipID)
+	if err != nil {
+		return err
+	}
+
+	if ip.IP.Server.Identifier == "" {
+		return nil
+	}
+
+	update.Address = ip.IP.Address
+	update.ID = ip.IP.ID
+	update.Organization = ip.IP.Organization
+	resp, err := s.PutResponse(ComputeAPI, fmt.Sprintf("ips/%s", ipID), update)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
```

Comment:
You have to add this code between `s.PutResponse` and `if err != nil`

```golang
if resp != nil {
    defer resp.Body.Close()
}
```

according to  http://devs.cloudimmunity.com/gotchas-and-common-mistakes-in-go-golang/#anameclose_http_resp_bodyaclosinghttpresponsebody